tests/test_molecule.py

```diff
@@ -167,7 +167,7 @@ def test_graph_from_atomic_coordinates_perception(
     m = copy.deepcopy(mol)
 
     delattr(m, "adjacency_matrix")
-    m.G = None
+    m.G = {}
```

Comment:
This is good, but we will need to add another test, similar to the one above (only if both backends are available), checking that the molecular graph is indeed re-generated for a new backend (and that at the end both are in the cache).

Reply:
Okay, I'll work on that.
Should I also test the interaction with molecule.strip and a change of backend?

Reply:
Molecule strip might be too much, but I think the change of backend needs to be tested.

Reply:
I'd add a very similar test to the one you added, and checking that the dictionary is populated by graphs from all backends.